lib/Runtime/Library/ForInObjectEnumerator.cpp

```diff
@@ -182,7 +207,32 @@ namespace Js
                 if (TestAndSetEnumerated(propertyId) //checks if the property is already enumerated or not
                     && (attributes & PropertyEnumerable))
                 {
-                    return currentIndex;
+                    bool propertyShadowed = false;
```

Comment:
Sorry, I guess I don't understand what your solution is
> Return the first prototype object and don't go up even if current prototype object is marked as no enumerable property.
Isn't that the same at not skip any prototype at all if there are any enumerable object in the prototype change?